File: filmy-app/app/src/main/java/rk/entertainment/filmy/utils/rvUtils/VerticalItemDecoration.java
package rk.entertainment.filmy.utils.rvUtils;


import android.content.Context;
import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import rk.entertainment.filmy.utils.UIUtils;

public class VerticalItemDecoration extends RecyclerView.ItemDecoration {

    private final int space;


    public VerticalItemDecoration(Context context) {
        this.space = UIUtils.dpToPx(8, context);
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent,
                               RecyclerView.State state) {
        outRect.right = space;
        if (parent.getChildAdapterPosition(view) == 0)
            outRect.left = space*2;
        if (parent.getChildAdapterPosition(view) != parent.getAdapter().getItemCount() - 1)
            outRect.bottom = space*2;
    }
}

File: filmy-app/app/src/main/java/rk/entertainment/filmy/utils/rvUtils/VerticalItemDecoration.kt
package rk.entertainment.filmy.utils.rvUtils

import android.content.Context
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration
import rk.entertainment.filmy.utils.UIUtils

class VerticalItemDecoration(context: Context) : ItemDecoration() {

    private val space: Int = UIUtils.dpToPx(8f, context)

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView,state: RecyclerView.State) {

        outRect.right = space
        if (parent.getChildAdapterPosition(view) == 0) outRect.left = space * 2
        if (parent.getChildAdapterPosition(view) != parent.adapter!!.itemCount - 1) outRect.bottom = space * 2
    }

}